exec/java-exec/src/main/java/org/apache/drill/exec/store/parquet/columnreaders/NullableVarLengthValuesColumn.java

```diff
@@ -69,11 +73,16 @@ protected boolean readAndStoreValueSizeInformation() throws IOException {
     if ( currDefLevel == -1 ) {
       currDefLevel = pageReader.definitionLevels.readInteger();
     }
-    if ( columnDescriptor.getMaxDefinitionLevel() > currDefLevel) {
+
+    if (columnDescriptor.getMaxDefinitionLevel() > currDefLevel) {
       nullsRead++;
-      // set length of zero, each index in the vector defaults to null so no need to set the nullability
-      variableWidthVector.getMutator().setValueLengthSafe(
-          valuesReadInCurrentPass + pageReader.valuesReadyToRead, 0);
+      // set length of zero, each index in the vector defaults to null so no
+      // need to set the nullability
+      if (variableWidthVector == null) {
```

Comment:
It took me a while to get some idea how this code is supposed to work.  It was not clear to me that only the last item in the list was accessed.  If you are certain that is the case, we don't need a list, but instead just a single value.